docs/rfc/007-endpoint-errors.md

```diff
@@ -0,0 +1,229 @@
+# [RFC] Conjure Endpoint Errors
+
+**Date:** 2020/03/19
+
+**Stakeholders:**
+* *Anyone that produces or consumes conjure-generated code.*
+
+* * *
+
+## **Context**
+
+Currently, errors can be defined in Conjure type definitions. These errors can then be thrown by services, and if a client 
+knows about them, they can be handled during runtime by clients. The exact handling of this mechanism may vary by
+the implementation (for example, in Java these are `ServicException`s, which become `RemoteException`s which can be caught
+by the client).
+
+There are multiple issues with this. As a consumer, you only know that an endpoint may throw an error by reading the 
+documentation or the implementation itself. As a producer, you are limited in the expressivity of what you can say 
+with your exception, as you have no guarantee that a consumer will handle your exception (or even read your documentation).
+
+This RFC introduces *declared errors* on Conjure endpoints. These errors represent real outcomes of a call to an endpoint, 
+and should be handled by consumers.
+
+## Existing Documentation
+
+### PRs for example implementation (in Java, using checked exceptions)
```

Comment:
Have you guys thought about how declared errors would be consumed in other languages? It would be interesting to see an idea of what the implementation for another commonly used language (TypeScript, Golang) would look like.